infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/KubernetesPluginsToolingValidator.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.wsplugins;
+
+import static java.lang.String.format;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import javax.inject.Singleton;
+import org.eclipse.che.api.workspace.server.spi.InternalInfrastructureException;
+import org.eclipse.che.api.workspace.server.wsplugins.model.PluginBase;
+
+@Singleton
+public class KubernetesPluginsToolingValidator {
+
+  // Pattern is from K8S Container class
+  private static final Pattern namePattern = Pattern.compile("[a-z0-9]([-a-z0-9]*[a-z0-9])?");
+
+  public void validatePluginNames(List<? extends PluginBase> plugins)
+      throws InternalInfrastructureException {
+    for (PluginBase plugin : plugins) {
+      final String formattedPluginName = plugin.getName().toLowerCase();
```

Comment:
you should check container name as well

Reply:
Check that `plugin.getName()` is not null

Reply:
I got why you don't check it here, and it's because of this field is missing by some plugins. It is needed to add this validation when it will be fixed. I'll add note that it should be added in the corresponding issue